pro/src/app/App/layouts/components/Header/components/HeaderDropdown/HeaderDropdown.tsx

```diff
@@ -65,25 +68,27 @@ export const HeaderDropdown = () => {
   const hideProfile =
     IN_STRUCTURE_CREATION_FUNNEL && offererOptions.length === 0
 
-  const selectedOffererId =
-    currentOffererId ??
-    getSavedOffererId(offererOptions) ??
-    (offererOptions.length > 0 ? offererOptions[0]?.value : '')
+  const selectedOffererName = currentOffererId
+    ? offererNames?.find(
+        (offererOption) => offererOption.id === currentOffererId
+      )
+    : undefined
 
-  const selectedOffererName = offererNames?.find(
-    (offererOption) => offererOption.id === Number(selectedOffererId)
-  )
-  const handleChangeOfferer = (newOffererId: string): void => {
+  const handleChangeOfferer = async (nextCurrentOffererId: string) => {
     // Reset offers stored search filters before changing offerer
     resetAllStoredFilterConfig()
 
-    // Updates offerer id in storage
-    if (storageAvailable('localStorage')) {
-      localStorage.setItem(SAVED_OFFERER_ID_KEY, newOffererId)
-    }
+    if (withSwitchVenueFeature) {
+      await dispatch(setCurrentOffererById({ nextCurrentOffererId })).unwrap()
+    } else {
+      // Updates offerer id in storage
+      if (storageAvailable('localStorage')) {
+        localStorage.setItem(SAVED_OFFERER_ID_KEY, nextCurrentOffererId)
+      }
 
-    // Hard refresh to homepage after offerer change
-    hardRefresh('/accueil')
+      // Hard refresh to homepage after offerer change
+      hardRefresh('/accueil')
+    }
```

Comment:
```suggestion
    await dispatch(setCurrentOffererById({ nextCurrentOffererId })).unwrap()
```

Simplification possible ?